pkg/controller/pxc/tls.go

```diff
@@ -311,3 +318,57 @@ func (r *ReconcilePerconaXtraDBCluster) createSSLManualy(cr *api.PerconaXtraDBCl
 	}
 	return nil
 }
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileTLSToggle(ctx context.Context, cr *api.PerconaXtraDBCluster) error {
+	if cr.CompareVersionWith("1.16.0") < 0 {
+		return nil
+	}
+
+	annotationTLSState, ok := cr.Annotations[naming.AnnotationTLS]
+	if !ok {
+		if err := k8s.AnnotateObject(ctx, r.client, cr, map[string]string{
+			naming.AnnotationTLS: string(naming.GetAnnotationTLSState(cr)),
+		}); err != nil {
+			return errors.Wrap(err, "failed to annotate")
+		}
+		return nil
+	}
+
+	if annotationTLSState == string(naming.GetAnnotationTLSState(cr)) {
+		return nil
+	}
+	clusterPaused, err := k8s.PauseCluster(ctx, r.client, cr)
+	if err != nil {
+		return errors.Wrap(err, "failed to pause cluster")
+	}
+	if !clusterPaused {
+		return nil
+	}
+	switch naming.AnnotationTLSState(annotationTLSState) {
+	case naming.AnnotationTLSStateEnabled:
+		if err := r.deleteCerts(ctx, cr); err != nil {
+			return errors.Wrap(err, "failed to delete tls secrets")
+		}
+	case naming.AnnotationTLSStateDisabled:
+	default:
+		return errors.Errorf("unknown value for %s annotation: %s", naming.AnnotationTLS, annotationTLSState)
+	}
+	crOrig := cr.DeepCopy()
+	cr.Spec.Unsafe.TLS = !*cr.Spec.TLS.Enabled
+	if cr.Spec.Unsafe.TLS != crOrig.Spec.Unsafe.TLS {
```

Comment:
if TLS is disabled without enabling the unsafe flag we'll return an error before (in `CheckNSetDefaults`) and won't enter this function. so I'm not sure if we need to do this check

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1844/commits/0bfb7c0ba1a692f8ebcf38fc4534f58712934010